### PR TITLE
Issue #3572 done

### PR DESCRIPTION
**Pull request by rahulshukla-webkul**
Issue #3572 done.

### CODE REVIEW

File: packages/Webkul/Product/src/Type/AbstractType.php [hunk 1]
```diff
@@ -924,4 +924,32 @@ public function isCartItemInactive(\Webkul\Checkout\Contracts\CartItem $item): b
...
+            $customerGroupPrices = $this->product->customer_group_prices()->get()->sortBy('qty')->values()->all();
+
+            $rulePrice = app('Webkul\CatalogRule\Helpers\CatalogRuleProductPrice')->getRulePrice($this->product);
+
+            if ($rulePrice && $rulePrice->price > $this->product->special_price) {
+                foreach ($customerGroupPrices as $key => $customerGroupPrice) {
+                    if ($key > 0) {
+                        $price = $this->getCustomerGroupPrice($this->product, $customerGroupPrice->qty);
+
+                        $save = (($this->product->price - $price)*100)/($this->product->price);
```
Comment (jitendra-webkul) [inline packages/Webkul/Product/src/Type/AbstractType.php:945]: add space before and after with binary operator

File: packages/Webkul/Product/src/Type/AbstractType.php [hunk 2]
```diff
@@ -924,4 +924,32 @@ public function isCartItemInactive(\Webkul\Checkout\Contracts\CartItem $item): b
...
+
+                        $offerLines[] = 'Buy' .' '. $customerGroupPrice->qty .' '.'for'.' '. core()->currency($price).' '. 'each and save'.'  '.$save.'%';
```
Comment (jitendra-webkul) [inline packages/Webkul/Product/src/Type/AbstractType.php:947]: add space before and after with binary operator

Comment (jitendra-webkul) [reply packages/Webkul/Product/src/Type/AbstractType.php:947]: Translation missing

**Comment by bhanu-webkul** [comment]
Still same issue 
 2nd offer is not visible to customer(velocity and default) and if customer is eligible for 2nd offer still 1st offer showing in cart(velocity theme) - https://prnt.sc/xdem3o

**Comment by bhanu-webkul** [comment]
In case there is 2 group price is added for same qty and if any customer group belong to both then on front end for that customer group only show 1 discount(higher one)
As of now- 
https://prnt.sc/xg7ne5

**Comment by bhanu-webkul** [comment]
getting exception on clicking catalog grid at admin end - https://prnt.sc/xqx1sh

**Comment by vaishaliwebkul** [comment]
@bhanu-webkul 
error is not coming from this PR, it exists in the master for which PR is already opened by @devansh-webkul https://github.com/bagisto/bagisto/pull/4526

**Comment by vaishaliwebkul** [comment]
@rahulshukla-webkul 
In reference to issue #4523 do the below updates also

- [ ]   when applying group pricing then the discount amount should also visible on the homepage as well as on the product page

https://prnt.sc/xrb0cw

https://prnt.sc/xrb2af

**Comment by vaishaliwebkul** [comment]
In reference to issue #4495, add the locale filter on product datagrid, velocity meta data, sliders grid also
https://prnt.sc/xrevqj
https://prnt.sc/xrexe0

**Review by vaishaliwebkul — APPROVED** [review]

**Pull request merged**